firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteSerializer.java

```diff
@@ -633,14 +633,25 @@ public com.google.firebase.firestore.core.Target decodeQueryTarget(QueryTarget t
   }
 
   StructuredAggregationQuery encodeStructuredAggregationQuery(
-      QueryTarget encodedQueryTarget, List<AggregateField> aggregateFields) {
+      QueryTarget encodedQueryTarget,
+      List<AggregateField> aggregateFields,
+      HashMap<String, String> aliasMap) {
     StructuredAggregationQuery.Builder structuredAggregationQuery =
         StructuredAggregationQuery.newBuilder();
     structuredAggregationQuery.setStructuredQuery(encodedQueryTarget.getStructuredQuery());
 
-    // We use a Set here to automatically remove duplicates.
-    Set<StructuredAggregationQuery.Aggregation> aggregations = new HashSet<>();
+    List<StructuredAggregationQuery.Aggregation> aggregations = new ArrayList<>();
+    HashSet<String> uniqueField = new HashSet<>();
+    int aliasID = 1;
     for (AggregateField aggregateField : aggregateFields) {
+      // deduplicate aggregateFields
+      final int count = uniqueField.size();
+      uniqueField.add(aggregateField.getAlias());
+      if (count == uniqueField.size()) {
```

Comment:
consider a comment for this block for future readers to understand that if the HashSet doesn't grow then the aggregation is a duplicate and we are not sending duplicate aggregates to the server.